pkg/connector/connector.go

```diff
@@ -127,20 +149,52 @@ func (c *Okta) Asset(ctx context.Context, asset *v2.AssetRef) (string, io.ReadCl
 	return "", nil, fmt.Errorf("not implemented")
 }
 
-func New(ctx context.Context, domain, apiToken string, syncInactiveApps bool) (*Okta, error) {
+func New(ctx context.Context, config map[string]any) (*Okta, error) {
+	var (
+		oktaClient *okta.Client
+		scopes     = []string{}
```

Comment:
scopes should be initialized to defaultScopes. Then farther down you can just have 
```
if provisioningEnabled {
    scopes = append(scopes, provisioningScopes...)
}
```
...without the else.

Also I think provisioning mode needs the default scopes anyway (at least for service mode).